packages/components/src/components/es-hole-puncher/es-hole-puncher.tsx

```diff
@@ -0,0 +1,71 @@
+import { Component, h, Element, Prop, Host } from '@stencil/core';
+
+/**
+ * Punches a hole through all parent shadow DOMs into the light DOM.
+ * Any slotted chlid will be moved into the light, and a chain of slots
+ * brings the slotted children back into position.
+ */
+@Component({
+    tag: 'es-hole-puncher',
+    shadow: true,
+})
+export class EsHolePuncher {
+    @Element() host!: Element;
+
+    /** The unique ID to use for the slot path */
+    @Prop() uniqueId!: string;
```

Comment:
I'm wondering if we should make the `uniqueId` optional and generate a `UUID` automatically if it's not provided.

Reply:
Yeah, I think you're right this should be handled internally. The only reason to pass your own id is so you can eyeball it in the DOM, so I changed `uniqueId` to `namePrefix`, so if you want that you don't have to think about keeping the id unique, it still handles that, just with a different prefix